org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -1219,16 +1219,12 @@ public static File fromUri(URI uri) {
 		return Paths.get(uri).toFile();
 	}
 
-	public static boolean hasCapability(Either<Boolean, ? extends Object> eitherCapability) {
-		if(eitherCapability != null) {
-			if (eitherCapability.isLeft()) {
-				return eitherCapability.getLeft();
-			} else {
-				return eitherCapability.getRight() != null;
-			}
-		} else {
+	public static boolean hasCapability(final @Nullable Either<Boolean, ? extends Object> eitherCapability) {
+		if(eitherCapability == null) {
 			return false;
 		}
+		return eitherCapability.isLeft()
```

Comment:
Just for sake of discussion and more about taste than correctness, I would have used `(either.isLeft() && either.getLeft()) || either.isRight())`, and this can be even simplified to `either.isRight() || either.getLeft()`.

Reply:
`either.isRight() || either.getLeft()` sounds good to me.